### ISSUE TITLE
Error connecting to service

### ISSUE THREAD

**Issue by 0-Il** [signalapp/Signal-Android#12985]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
After importing my backup and registering my phone number, I enter my PIN, but get `Error connecting to service`

### Steps to reproduce
- Import backup, register phone number
- Enter PIN code

**Actual result:** After entering my PIN code, I get `Error connecting to service` and cannot complete the registration.
**Expected result:** After entering my PIN code, the registering process should complete.

### Device info

**Device:** Huawei Honor 6A DLI-TL20
**Android version:** 7.0
**Signal version:** 6.21.3

### Link to debug log

https://debuglogs.org/android/6.21.3/9d4541d1aafb6fffa828ea5ad367064969537b41a06d323a14bd55c29ccadc1c

**Comment by 0-Il** [comment]
Reopening this as I suspect that my current issue is related to the above one.
I can't change my number, the app just returns me to the previous screen.
Debug log: https://debuglogs.org/android/6.23.5/ccda5c49f3bfbfa4ba0b75f2325747d012d56105a165830aa700b10fadb0a764

**Comment by 0-Il** [comment]
Pinging @clark-signal

**Comment by 0-Il** [comment]
Just tried again and have a different error: when trying to change the number, it says `Unable to request a verification code. Please check network connection and try again`

Debug log: https://debuglogs.org/android/6.23.5/0e4ec193fa159683c07baf55aab567ee382e978ccee4ef20f001f4298a8a25d9

**Comment by clauz9** [comment]
I think.. this got fixed... 

the `error connecting to service message` was an erroneous message, they  changed it, to a dialog box, that says you need to wait a certain amount of time...

**Comment by greyson-signal** [comment]
@bohwaz Can you share a log? There's a secret way to get a log during registration by tapping on any header text 8 times.

**Comment by bohwaz** [comment]
@greyson-signal thank you but I solved the issue by uninstalling the app and reinstalling it.

**Comment by ph818** [comment]
I get this error as well. 

https://debuglogs.org/android/7.45.3/ae8c6591e3168d11b13984e23e0519d6806bc88673038933b6b73663c9d86cce.

My context is a little complicated. I have two phones, identical make/model. In the past year, I moved signal successfully from phone A to phone B. Phone B recently died.

On Phone A, I went to reactivate Signal. I selected "cancel activation on new device and keep using signal on this one" or similar. 

Now I'm prompted to "reregister device." The app says it will send my number a verification code, but never sends one and never asks for one. It asks for my pin, but when I put it in, it says 

"Error connecting to service."

If I click Skip, it asks for a verification code again, and this time sends one and auto-inputs it into the field (weird... How does it have permission to read my texts?)

Then it asks for my pin again, and this time it works. Finally it asks me to update my pin to complete registration.
